### ISSUE TITLE
Trouble creating USB Installer

### ISSUE THREAD

**Issue by richb-hanover** [ameeno/Dell-E7440-Hackintosh#37]
Thanks for creating this guide. I'm excited to try it out. I have a couple questions:

1. (Slightly OT for this repo) I am stuck at the step of copying the BaseSystem.dmg on to the USB drive. I posted a report at https://github.com/midi1996/hackintosh-internet-install-gitbook/issues/11  
   Do you have any thoughts about getting this going? workarounds?

2. In your [Creating the USB Installer](https://github.com/ameeno/Dell-E7440-Hackintosh/blob/master/README.md) step, you say to place the CLOVER files on a separate USB drive. But the steps (above) seem to create two partitions on the USB drive - one 200MByte FAT partition named CLOVER, and a second that ends up with the BaseSystem.dmg contents. Do I need *yet another* USB drive? Or will these two partitions on that single drive suffice? 

Thanks.

**Comment by simonegraziano** [comment]
You need a 2nd usb drive because after first boot (using the clover boot loader on 1st usb drive) you need to copy CLOVER files modded by ameeno to boot loader section (EFI Partition) on your hard drive. (You need mounting EFI partition if its is not visible (not mounted) )

**Comment by jtitley** [comment]
Seems the ethernet driver included doesn't work. When I revert to a previous version it will work, but only if it isn't plugged in.

In any event, I've been trying to follow these instructions, and it's now my third attempt at installing Mojave. It's not clear how this USB swap is supposed to work. When there is a reboot, it just goes back to the recovery screen. The instructions mention follow the same instructions to install Clover on the hard drive but there is no option to do this. Clover is installed on the USB key from Windows. I don't understand how this is supposed to work.

I have the exact same model and bios revision as the author.

**Comment by richb-hanover** [comment]
Yes, the instructions pretty little vague about exactly what happens at this critical juncture of the installation process. (This is a generic fault of most of the Hackintosh documentation. I am considering writing an update that explains this rough spot. I found it helpful to watch several of Technolli's Youtube videos to get an overview of the installation process.)

Here's what (I think) I did when I successfully installed. (I say "I think this is what I did..." because I wasn't taking careful notes...)

* Plug the USB Drive (with two partitions - Clover & macOS Installer) into the target
* Boot, set boot options, let Clover start up
* Select the "install from macOS option" and let it get to the main installer screen
* Run Disk Utility to format the internal hard drive (APFS, GUID Partition Table)
* Quit Disk Utility, and you're back to the macOS Installer "home page"
* Start the macOS installer in the obvious way

The macOS installer needs to reboot a few times. Each time, as you note, you get dumped back into the Clover menu. I am pretty sure I simply chose the "boot from macOS" and the installation process picked up from that point on. It took three or maybe four reboots before macOS was fully installed.

At that point, it was booting (getting the EFI info) from the USB drive, but then I would tell Clover to boot from the internal drive, and macOS would start fully.

Then I followed some instructions I found (again, vague, and I don't have a link to exactly what I did) to run Clover Configurator to mount the internal and USB drive EFI volumes. I then copied across the stuff from the Clover partition of the USB drive to the EFI partition of the internal drive.

**Comment by jtitley** [comment]
Yeah, I get where you are going with this and what you say is what I sort of thought. Interestingly, there was no option for APFS but it seems the installer converted it anyhow. I got to where Apple is booting from the hard drive. But what happens now is that it hangs at the apple logo. I turned on debugging, and it shows a kernel panic.

I do not understand how where to go from here. AFAIK I have the same laptop, same BIOS and the original other. The only difference I can see is the downloaded recovery is from 2020. And maybe the Mojave image differs slightly from when this was written.

**Comment by jtitley** [comment]
I've determined I was getting a kernel panic about APFS. apfs volume keybag, err = 2. I don't know why.

I installed High Sierra instead, and that worked without issue. So something about Mojave doesn't work anymore.

Good enough for me.

**Comment by richb-hanover** [comment]
Sounds good. I was about to suggest you retry the install. And you did. Good luck!

**Comment by Aristocrat-jt** [comment]
Weird, but when I installed the wifi card (originally it wasn't installed) and tried a fresh install again, it worked flawlessly. I then did an in-place Mojave upgrade from High Sierra and that worked flawlessly. Sound, mouse, keyboard, everything. Plus, I have dual boot to Windows 10 in case something really doesn't work.

Maybe it's because I updated some drivers and updated clover from this release (there are a few drivers here that are out of date). I don't know. In any event, the only thing I need to try is HDMI out. Once that works, this will be my new daily machine.

**Comment by jtitley** [comment]
Weird, but when I installed the wifi card (originally it wasn't installed) and tried a fresh install again, it worked flawlessly. I then did an in-place Mojave upgrade from High Sierra and that worked flawlessly. Sound, mouse, keyboard, everything. Plus, I have dual boot to Windows 10 in case something really doesn't work.

Maybe it's because I updated some drivers and updated clover from this release (there are a few drivers here that are out of date). I don't know. In any event, the only thing I need to try is HDMI out. Once that works, this will be my new daily machine.
